source/plugin/modularization.rst

```diff
@@ -0,0 +1,62 @@
+===================
+Code Modularization
+===================
+
+When writing Sponge plugins, you may find that you need to modularize your code. Maybe you want to split your core into
+another project so that others can make use of it, or maybe you want to utilize a service another plugin provides. With
+Sponge, gone are the days of shading entire plugins into your plugin.
+
+Problem 1: Providing Good Service
+=================================
+
+Let's say you want to provide some awesome service for ``WarpService``. You wrote ``AwesomeWarpService`` that uses SQL
```

Comment:
I'm not sure I understand why most of this paragraph is necessary. Correct me if I'm wrong, but the only thing you are showing is that this `WarpService` class cannot be found by Sponge. The rest of the details about SQL and flat files and whatnot seems a bit extraneous.